src/highlight.js

```diff
@@ -845,8 +844,7 @@ https://highlightjs.org/
 
     block.innerHTML = result.value;
     block.className = buildClassName(block.className, language, result.language);
-    if(options.addLanguageAttr)
-       block.setAttribute('data-language',result.language);
+    block.setAttribute('data-language', block.getAttribute('data-language') || result.language);
```

Comment:
> I think we only want to do this if the attribute doesn't already exist though... i don't see that you added any code to check that?

Well, here I meant to set the attribute value to `result.language` except the attribute already has a value, in that case the attribute will remain with the initially defined value.
Maybe you mean something more like:
`if(!block.hasAttribute('data-language')) block.setAttribute('data-language', result.language);`
What do you think?